primitives/src/vrf.rs

```diff
@@ -0,0 +1,111 @@
+use num::rational::BigRational;
+use num::ToPrimitive;
+use parity_scale_codec::{Decode, Encode};
+use sp_arithmetic::per_things::{PerThing, Perbill, Perquintill};
+use sp_runtime_interface::pass_by::PassByCodec;
+use sp_runtime_interface::runtime_interface;
+
+mod model {
+	use super::*;
+
+	/// S is a hyperparameter representing participation rate.
+	/// The higher the value, the higher the chances of being sampled.
+	/// R is the prover's relative stake. the output is proportional to the stake.
+	pub fn model(s: Perquintill, r: Perquintill) -> Perquintill {
```

Comment:
From the Jira description: `sample_size`: [0,1) and `weight` as a stake ratio: [0,1). 

Sounds like this function should explicitly check the range values and panic.  I am reluctant to disregard these constraints if they are documented as required behavior.